ndt7/results/metadata.go

```diff
@@ -14,23 +12,13 @@ type metadata map[string]string
 // serverKeyRe is a regexp that matches any server related key.
 var serverKeyRe = regexp.MustCompile("^server_")
 
-// initMetadata initializes |*meta| from |localEpnt|, |remoteEpnt|, the |values|
-// provided using the query string, and the |subtest| name. Returns an error
-// on failure.
-func initMetadata(m *metadata, localEpnt, remoteEpnt, uuid string, values url.Values, subtest string) {
+// initMetadata initializes |*meta| from |values| provided from the original
+// request query string.
+func initMetadata(m *metadata, values url.Values) {
 	for k, v := range values {
 		if matches := serverKeyRe.MatchString(k); matches {
 			continue // We MUST skip variables reserved to the server
 		}
-		if len(v) != 1 {
-			continue // We SHOULD ignore duplicate keys
-		}
```

Comment:
Why are you removing this code? At least you should check for `len(v) > 0` because in principle `v[0]` can panic and this string here is provided by the client.

Reply:
Can you provide an example where v is empty? See for example: https://play.golang.org/p/NXba6pTmCNa